src/dune_engine/dune_project.ml

```diff
@@ -908,6 +924,13 @@ let parse ~dir ~lang ~file ~dir_status =
           let root = dir in
           let file_key = File_key.make ~name ~root in
           let dialects =
+            let dialects =
+              match
+                File_key.Map.find explicit_extensions Melange_syntax.name
```

Comment:
Not sure why you used the File_key map. I switched it to a regular string map.

Reply:
I think it was because Merlin reported it as the type, for appearing later in the file than StringMap. Appreciate the catch + fix